Review IPv4/IPv6 installation requirements (#4213)
theforeman/foreman-documentation: guides/common/modules/ref_ipv6-and-ipv4-requirements.adoc

[id="ipv6-and-ipv4-requirements"]
= IPv6 and IPv4 requirements

You can install {Project} in an IPv4 network or in an IPv6 network.
ifdef::satellite[]

> Dual-stack {Project} installation that uses both IPv4 and IPv6 is not supported.

Reviewer: ```suggestion
Dual-stack {Project} installations that use both IPv4 and IPv6 are not supported.
```

alternative:

```suggestion
Performing a dual-stack {Project} installation that uses both IPv4 and IPv6 is not supported.
```

Or even just "_A_ dual-stack ...". Your call.

endif::[]

The following requirements apply to installations in an IPv4 network:

* Ensure an IPv6 loopback is configured on the base system.
The loopback is typically configured by default.
Do not disable it.
* Do not disable IPv6 in kernel by adding the `ipv6.disable=1` kernel parameter.

The following requirements apply to installations in an IPv6 network:

ifdef::satellite[]
* Deploy an external HTTP proxy server that supports both IPv4 and IPv6.
This is required because Red Hat Content Delivery Network distributes content only over IPv4 networks, therefore you must use this HTTP proxy to pull content into the {Project} on your IPv6 network.

> You must configure {Project} to use this dual-stack (supporting both IPv4 and IPv6) HTTP proxy as the default HTTP proxy.

Reviewer 2: Just a heads up that this requirement will disappear in the next few months. The requirement came from the RH CDN being IPv4-only, but they will make that a dual stack deployment so IPv6-only deployments don't need it. https://issues.redhat.com/browse/SAT-27425 if you want to follow this.

After that is available, we will have the same note as for upstream: users may still pull in custom content that is IPv4-only.

Author: Do you know which version will have the fix? The Jira doesn't have that information.

Reviewer 2: I'm not sure that is public information so see the private message.

endif::[]
ifndef::satellite[]
* If you rely on content from IPv4-only networks, use an external dual-stack HTTP proxy.
Configure {Project} to use this dual-stack (supporting both IPv4 and IPv6) HTTP proxy as the default HTTP proxy.

> endif::[]

Reviewer 3: This probably should not be rendered for foreman-el/foreman-deb, as there is no concept of "content" there.

* {Project} does not support configuring an HTTP proxy using a direct IPv6 address.
Instead, configure the HTTP proxy with a FQDN that resolves to the IPv6 address.

> Using an IPv6 address as the HTTP proxy URL causes it to fail.

Author: I'll keep this in for now, then, and please let me know if you find out it's safe to remove.

> Using an IPv6 address as the HTTP proxy URL causes it to fail.

Author: Hi @ianballou, I see you approved the linked Katello PR. Do you know if it's safe to remove this bullet point?

> Using an IPv6 address as the HTTP proxy URL causes it to fail.

Reviewer: ```suggestion
```

I would drop this line; it does not add any value after line 27+28.

> If you also intend to provision hosts in an IPv6 network, consider the following requirements:

Reviewer: ```suggestion
The following requirements apply to provisioning hosts in an IPv6 network:
```

suggestion (non-blocking): Align with lines 9 and 16.

Author: Applied, except that I kept "If you intend to [...]" at the beginning. That's import so that users know they can skip this if they don't meet the condition.

* Deploy an external DHCPv6 server and configure it manually to communicate with the network boot process and to manage IP address assignment because {Project} cannot integrate with a DHCPv6 server and manage its configuration.
For more information about DHCPv6 server configuration, see {ProvisioningDocURL}options-in-unmanaged-dhcpv6[Options in unmanaged DHCPv6] in _{ProvisioningDocTitle}_.

> * Although {Project} provisioning templates include IPv6 support for PXE and HTTP (iPXE) provisioning, the only tested and certified provisioning workflow is the UEFI HTTP Boot provisioning.

Reviewer: I suggest that you reword this from "although X, so Y" to "You use UEFI HTTP Boot provisioning." to make it sound more like a requirement. Feel free to add the remaining information into an addional line. Right now, this sounds more like a suggestion to me than a hard technical requirement.

Author: Actually, I think we can merge this with the bullet point below.
